android/java/org/chromium/chrome/browser/brave_news/BraveNewsBottomSheetDialogFragment.java

```diff
@@ -0,0 +1,126 @@
+
+    public static BraveNewsBottomSheetDialogFragment newInstance() {
+        return new BraveNewsBottomSheetDialogFragment();
+    }
+
+    public void setController(BraveNewsController controller) {
+        mBraveNewsController = controller;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setStyle(STYLE_NORMAL, R.style. AppBottomSheetDialogTheme);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        if (getArguments() != null) {
+            url = getArguments().getString("url");
+            publisherId = getArguments().getString("publisherId");
+            publisherName = getArguments().getString("publisherName");
+        }
+        View view = inflater.inflate(R.layout.brave_news_card_menu, container, false);
+        return view;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        mContext = context;
+    }
+
+    @Override
+        public void onDetach() {
+        super.onDetach();
+        mContext = null;
+    }    
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        TextView title = view.findViewById(R.id.item_menu_title_text);
+        Button newTab = view.findViewById(R.id.new_tab);
+        Button privateTab = view.findViewById(R.id.new_private_tab);
+        Button disable = view.findViewById(R.id.disable_content);
+
+        title.setText(url);
+        StringBuilder disableText = new StringBuilder();
+        disableText.append(getResources().getString(R.string.brave_news_disable_content));
+        disableText.append(" ");
+        disableText.append(publisherName);
+        disable.setText(disableText);
+
+        newTab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                TabUtils.openUrlInNewTab(false, url);
+                dismiss();
+            }
+        });
+
+        privateTab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                TabUtils.openUrlInNewTab(true, url);
+                dismiss();
+            }
+        });
+
+        disable.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mBraveNewsController != null){
+
+                    SharedPreferencesManager.getInstance().writeBoolean(BravePreferenceKeys.BRAVE_NEWS_CHANGE_SOURCE, true);
+                    mBraveNewsController.setPublisherPref(publisherId, UserEnabled.DISABLED);
+                }
+                dismiss();
+                Toast.makeText(mContext, publisherName + " disabled", Toast.LENGTH_SHORT).show();
```

Comment:
what is that for?

Reply:
Removes the news source from the fetch list. 
- sets a pref to signal there was a source change that is caught in BraveNewTabLayout
- sets the UserEnabled.DISABLED for that source
- shows a Toast to notify the user

Reply:
shouldn't `disabled` be translated?

Reply:
refactored